### PR TITLE
Add data from API to About modal

### PR DESCRIPTION
**Pull request by ZitaNemeckova**
Adds data from API to About modal. Thanks @himdel for crash course in SSUI :)

Works with https://github.com/ManageIQ/manageiq-ui-self_service/pull/119

@AparnaKarve please have a look, thanks :)

### CODE REVIEW

File: client/app/components/about-modal/about-modal.controller.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+(function() {
+    'use strict';
+angular.module('app.components').controller('ModalCtrl', function ($scope, ServerInfo) {
```
Comment (himdel) [inline client/app/components/about-modal/about-modal.controller.js]: broken formatting ;)

**Comment by AparnaKarve** [comment]
@ZitaNemeckova This looks really great for the first pass 👍  

Like we discussed earlier, we preferably -
a)  do not want to reference  `ng-controller` in the html 
b) do not want to use`$scope` in the controller code.

So how about we add a few more commits in this PR to achieve that? 😄 

/cc @himdel

**Comment by AparnaKarve** [comment]
@ZitaNemeckova Great progress! 
The last 2 commits are exactly what I was looking for.

I'm more or less OK with the implementation so far, but maybe we should strive for consistency now.
... and by that I mean if you look at other components that act as directives/controllers in `client/app/components`, you'll see that they -
a) do not make the `.controller()` call
b) only make the `.directive()` call 
c) They have 2 main `ngInject-able` functions one for the `directive` code and the other for the `controller` code 

Can you add a commit that will do the above?
Thanks.

**Comment by AparnaKarve** [comment]
Check list items in https://github.com/ManageIQ/manageiq-ui-self_service/pull/119 need to be addressed as well.

File: client/index.html [hunk 2]
```diff
@@ -27,7 +27,6 @@
   <link rel="stylesheet" href="/bower_components/bootstrap-datepicker/dist/css/bootstrap-datepicker3.css" />
   <link rel="stylesheet" href="/bower_components/bootstrap-select/dist/css/bootstrap-select.css" />
   <link rel="stylesheet" href="/bower_components/bootstrap-switch/dist/css/bootstrap3/bootstrap-switch.css" />
-  <link rel="stylesheet" href="/bower_components/bootstrap-touchspin/src/jquery.bootstrap-touchspin.css" />
```
Comment (himdel) [inline client/index.html]: It reflects the current dependencies, not caused by changes in this PR, but ..well, current :).

Related to https://github.com/ManageIQ/manageiq-ui-self_service/issues/81

File: client/app/services/language.service.js [hunk 3]
```diff
@@ -100,6 +100,8 @@
 
       var code = getLocale(data);
       setLocale(code);
+      
+      return data;
```
Comment (himdel) [inline client/app/services/language.service.js]: `Language.onReload` did not return the original data, so could not be chained [here](https://github.com/ManageIQ/manageiq-ui-self_service/pull/127/files#diff-b39a4f7482697f7f1a49c46b01c64d1fR92)

File: client/index.html (hunk 2 above)
Comment (AparnaKarve) [inline client/index.html]: In any case, we don't need these changes

File: client/app/services/language.service.js (hunk 3 above)
Comment (AparnaKarve) [inline client/app/services/language.service.js]: Seems like this change needs a special mention in a separate commit.

**Comment by AparnaKarve** [comment]
@ZitaNemeckova ping

**Comment by ZitaNemeckova** [comment]
@miq-bot remove_label wip

@AparnaKarve please review :)

File: .jshintrc [hunk 4]
```diff
@@ -23,7 +23,7 @@
   "undef"         : true,     // true: Require all non-global variables to be declared (prevents global leaks)
   "unused"        : false,    // true: Require all defined variables be used
   "strict"        : false,    // true: Requires all functions run in ES5 Strict Mode
-  "maxparams"     : 10,       // {int} Max number of formal params allowed per function
+  "maxparams"     : 12,       // {int} Max number of formal params allowed per function
```
Comment (AparnaKarve) [inline .jshintrc]: @ZitaNemeckova I'm not so sure about this change.

Are we going to bend this rule every time the ngInject params cap reaches the max value just to make the tests pass?
Soon enough, we would have a requirement for 13 and then we'll change it 13. 
In that case why even have this rule?

@chriskacerguis @AllenBW What do you think about this?

File: client/index.html (hunk 2 above)
Comment (AparnaKarve) [inline client/index.html]: @ZitaNemeckova This is still not addressed.

File: .jshintrc (hunk 4 above)
Comment (chriskacerguis) [inline .jshintrc]: I'm in agreement with @AparnaKarve, we shouldn't be bending the fils for a change.

Comment (AllenBW) [inline .jshintrc]: yeah its a b to get it to work sometimes, but sadly the pain must be felt 😿 

sorry for delay in review @AparnaKarve, looking at this now 👍

File: client/app/components/about-modal/about-modal.controller.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
+(function() {
+  'use strict';
+    
+  angular.module('app.components')
+    .directive('aboutModal', function() {
+      return {
+        templateUrl: "app/components/about-modal/about-modal.html",
+        controller: ModalCtrl,
```
Comment (AllenBW) [inline client/app/components/about-modal/about-modal.controller.js]: gonna be a stickler and request we call this `ModalController` (eslint is gonna bark in the near future anyways)

File: client/app/components/about-modal/about-modal.controller.js [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+        controllerAs: "info",
+      };
+    });
+  /** @ngInject */
+  function ModalCtrl(ServerInfo) {
+    var vm = this;
+    this.additionalInfo = "";
```
Comment (AllenBW) [inline client/app/components/about-modal/about-modal.controller.js]: should this be `vm.additionalInfo`?  (same with everything below)

File: client/app/components/about-modal/about-modal.controller.js [hunk 7]
```diff
@@ -0,0 +1,35 @@
+(function() {
+  'use strict';
+    
+  angular.module('app.components')
+    .directive('aboutModal', function() {
+      return {
+        templateUrl: "app/components/about-modal/about-modal.html",
+        controller: ModalCtrl,
+        controllerAs: "info",
```
Comment (AllenBW) [inline client/app/components/about-modal/about-modal.controller.js]: app convention is to use `controllerAs: "vm"`

File: client/app/components/about-modal/about-modal.html [hunk 8]
```diff
@@ -0,0 +1,6 @@
+<a href="#"  ng-click="info.open()" >
+    About
+</a>
+<div pf-about-modal is-open="info.isOpen" on-close="info.onClose()" additional-info="info.additionalInfo"
```
Comment (AllenBW) [inline client/app/components/about-modal/about-modal.html]: so this would be `is-open="vm.isOpen()"` (probs want params on this)

same applies to other instances of `info.xxx`

File: client/app/layouts/_application.sass [hunk 9]
```diff
@@ -1,3 +1,15 @@
+// Product overrides of Boostrap & Patternfly variables
```
Comment (AllenBW) [inline client/app/layouts/_application.sass]: on snap, we DONT have a file for all misc vendor overrides. this needs to happen, not in this commit but **thank you** for mentioning this as an override here

File: client/app/components/about-modal/about-modal.html [hunk 10]
```diff
@@ -0,0 +1,6 @@
+<a href="#"  ng-click="info.open()" >
+    About
```
Comment (AllenBW) [inline client/app/components/about-modal/about-modal.html]: can we add the `translate` attribute to this element? or do something like `{{ About | translate}}`

point being, all hardcoded text gotta support translations

File: client/app/layouts/application.html [hunk 11]
```diff
@@ -28,6 +28,26 @@
...
         </li>
+        <li class="dropdown">
+          <a class="dropdown-toggle nav-item-iconic" id="aboutModal" data-toggle="dropdown" aria-haspopup="true" aria-expanded="true">
+            <span class="fa pficon-help"></span>
+            <span class="caret"></span>
+          </a>
+          <ul class ="dropdown-menu" aria-labelledby="aboutModal">
+            <li>
+              <a href="/support/index?support_tab=about">
+                Documentation
```
Comment (AllenBW) [inline client/app/layouts/application.html]: translate me! <3

File: client/app/services/server-info.service.js [hunk 12]
```diff
@@ -0,0 +1,25 @@
+(function() {
+  'use strict';
+
+  angular.module('app.services')
+    .service('ServerInfo', ServerInfo);
+
+  /** @ngInject */
+  function ServerInfo($q) {
```
Comment (AllenBW) [inline client/app/services/server-info.service.js]: `var vm = this;` <3

File: client/app/layouts/_application.sass (hunk 9 above)
Comment (ZitaNemeckova) [inline client/app/layouts/_application.sass]: @AllenBW I can't take credit for it :) , this comment comes from @epwinchell PR #119 that I used as base for my PR.

File: .jshintrc (hunk 4 above)
Comment (ZitaNemeckova) [inline .jshintrc]: @AparnaKarve  @chriskacerguis  @AllenBW I disabled that test as I cannot to my knowledge remove anything from `function StateController(...)`. @AparnaKarve was right that bending it over and over again makes it pointless anyway. If there is a nicer way to work around that test, please let me know :)

**Comment by ZitaNemeckova** [comment]
@AllenBW thanks for review it was really helpful :) All your comments should be addressed.

File: .jshintrc (hunk 4 above)
Comment (chriskacerguis) [inline .jshintrc]: @AllenBW can you help @ZitaNemeckova here?

Comment (AllenBW) [inline .jshintrc]: @ZitaNemeckova eeeeyyyoooooo can you send me the offending line? (link to it in github)

Comment (AparnaKarve) [inline .jshintrc]: @AllenBW Don't know if you already received the github link to the offending line or not, but here it is anyway --
https://github.com/ZitaNemeckova/manageiq-ui-self_service/blob/7625be60e7942d5b5e15ba57919e6bbd2443553d/client/app/states/login/login.state.js#L29

And you know what I just noticed? `CollectionsApi` ngInject is not being used at all. Does that mean we caught a lucky break? 😄 
We could just delete `CollectionsApi` and then the params count goes back to 10, which takes care of the offense (for now)

File: client/app/components/about-modal/about-modal.controller.js [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+        controller: ModalController,
+        controllerAs: "vm",
+      };
+    });
+  /** @ngInject */
+  function ModalController(ServerInfo) {
+    var vm = this;
+    vm.additionalInfo = "";
+    vm.copyright = __("Copyright (c) 2016 ManageIQ. Sponsored by Red Hat Inc.");
+    vm.imgAlt = "ManageIQ logo";
```
Comment (Fryguy) [inline client/app/components/about-modal/about-modal.controller.js]: This feels like it will break skinning.  Can you please verify that these strings are changed when productized with CloudForms?

File: .jshintrc (hunk 4 above)
Comment (AllenBW) [inline .jshintrc]: @ZitaNemeckova totally could remove `CollectionsApi`!  🎨 🎊 we also don't use `Notification` any longer, (now its `EventNotification` and if you need more i could put some work in to get rid of `$state` and `$rootScope`

File: .jshintrc [hunk 14]
```diff
@@ -23,7 +23,7 @@
   "undef"         : true,     // true: Require all non-global variables to be declared (prevents global leaks)
   "unused"        : false,    // true: Require all defined variables be used
   "strict"        : false,    // true: Requires all functions run in ES5 Strict Mode
-  "maxparams"     : 10,       // {int} Max number of formal params allowed per function
```
Comment (chriskacerguis) [inline .jshintrc]: This needs to change back to the original value

File: client/app/layouts/application.html [hunk 15]
```diff
@@ -28,6 +28,28 @@
...
         </li>
+        <li class="dropdown">
+          <a class="dropdown-toggle nav-item-iconic" id="aboutModal" data-toggle="dropdown" aria-haspopup="true" aria-expanded="true">
+            <span class="fa pficon-help"></span>
+            <span class="caret"></span>
+          </a>
+          <ul class ="dropdown-menu" aria-labelledby="aboutModal">
+            <li>
+              <a href="/support/index?support_tab=about">
+                <span translate>
```
Comment (chriskacerguis) [inline client/app/layouts/application.html]: I don't think this is how we want to do translates anymore.  See https://github.com/ManageIQ/manageiq-ui-self_service/pull/182 for good examples.

File: client/app/components/about-modal/about-modal.html [hunk 16]
```diff
@@ -1,7 +1,5 @@
-<a href="#"  ng-click="vm.open()" >
-    <span translate>
-      About
-    </span>
+<a href="#"  ng-click="vm.open()" translate>
```
Comment (chriskacerguis) [inline client/app/components/about-modal/about-modal.html]: @ZitaNemeckova it should look like this:

`
<a href="#"  ng-click="vm.open()">{{ 'About' | translate}}</a>
`

File: client/app/layouts/application.html [hunk 17]
```diff
@@ -35,10 +35,8 @@
           </a>
           <ul class ="dropdown-menu" aria-labelledby="aboutModal">
             <li>
-              <a href="/support/index?support_tab=about">
-                <span translate>
-                  Documentation
-                </span>
+              <a href="/support/index?support_tab=about" translate>
```
Comment (chriskacerguis) [inline client/app/layouts/application.html]: Same here:
`
<a href="/support/index?support_tab=about">{{ 'Documentation' | translate}}</a>
`

File: client/app/components/about-modal/about-modal.html (hunk 16 above)
Comment (ZitaNemeckova) [inline client/app/components/about-modal/about-modal.html]: Seems inconsistent with #182 to me... @mzazrivec can you elaborate, please?

Comment (mzazrivec) [inline client/app/components/about-modal/about-modal.html]: In this case, both variants are possible.

```html
<a translate>
...
</a>
```

wouldn't work only if the text of the `a` element would contain another html markup. In this case, it only contains the 'About' string, so it will work just fine.

File: client/app/components/about-modal/about-modal.controller.js (hunk 13 above)
Comment (chriskacerguis) [inline client/app/components/about-modal/about-modal.controller.js]: @ZitaNemeckova ping :)

File: client/app/components/about-modal/about-modal.html [hunk 18]
```diff
@@ -0,0 +1,6 @@
...
+  About
```
Comment (chriskacerguis) [inline client/app/components/about-modal/about-modal.html]: Should be translated per the other way (see other comments), if nothing else to keep a common style.

File: client/app/layouts/application.html [hunk 19]
```diff
@@ -28,6 +28,26 @@
...
         </li>
+        <li class="dropdown">
+          <a class="dropdown-toggle nav-item-iconic" id="aboutModal" data-toggle="dropdown" aria-haspopup="true" aria-expanded="true">
+            <span class="fa pficon-help"></span>
+            <span class="caret"></span>
+          </a>
+          <ul class ="dropdown-menu" aria-labelledby="aboutModal">
+            <li>
+              <a href="/support/index?support_tab=about" translate>
+                Documentation
```
Comment (chriskacerguis) [inline client/app/layouts/application.html]: See other comments about translation :)

**Comment by chriskacerguis** [comment]
@ZitaNemeckova looks like also a conflict and a code climate problem.

File: client/app/components/about-modal/about-modal.controller.js (hunk 13 above)
Comment (himdel) [inline client/app/components/about-modal/about-modal.controller.js]: Verified that the last version of the changes works with skinning (after https://github.com/ManageIQ/manageiq-ui-self_service/pull/192 gets in so that skinning actually works again :)).

@Fryguy I'm not sure about the copyright info, if we want the skins to be able to override that as well, this will need changing, but I suggest we do that in a separate PR, ..dependencies.

**Comment by chriskacerguis** [comment]
Nice job @ZitaNemeckova !!!

**Comment by ZitaNemeckova** [comment]
@chriskacerguis Thank you :)

**Comment by chessbyte** [comment]
Euwe Backport conflict:
```sh
$ git cherry-pick -e -x -m 1 66af10e  
error: could not apply 66af10e... Merge pull request #127 from ZitaNemeckova/about_modal
hint: after resolving the conflicts, mark the corrected paths
hint: with 'git add <paths>' or 'git rm <paths>'
hint: and commit the result with 'git commit'

$ git status
On branch euwe
Your branch is up-to-date with 'upstream/euwe'.
You are currently cherry-picking commit 66af10e.
  (fix conflicts and run "git cherry-pick --continue")
  (use "git cherry-pick --abort" to cancel the cherry-pick operation)

Changes to be committed:

	new file:   client/app/components/about-modal/about-modal.controller.js
	new file:   client/app/components/about-modal/about-modal.html
	modified:   client/app/layouts/_application.sass
	modified:   client/app/layouts/application.html
	modified:   client/app/services/language.service.js
	new file:   client/app/services/server-info.service.js
	modified:   client/app/states/login/login.state.js
	new file:   client/assets/images/bg-modal-about-pf.png
	modified:   client/index.html

Unmerged paths:
  (use "git add <file>..." to mark resolution)

	both modified:   client/app/config/authorization.config.js

$ git diff
```

```diff
diff --cc client/app/config/authorization.config.js
index d14a88c,9f0ad21..0000000
--- a/client/app/config/authorization.config.js
+++ b/client/app/config/authorization.config.js
@@@ -57,10 -57,10 +57,17 @@@
    }
  
    /** @ngInject */
++<<<<<<< HEAD
 +  function init($rootScope, $state, Session, $sessionStorage, logger, Language) {
 +    $rootScope.$on('$stateChangeStart', changeStart);
 +    $rootScope.$on('$stateChangeError', changeError);
 +    $rootScope.$on('$stateChangeSuccess', changeSuccess);
++=======
+   function init($rootScope, $state, Session, $sessionStorage, logger, Language, ServerInfo) {
+     var unregisterStart = $rootScope.$on('$stateChangeStart', changeStart);
+     var unregisterError = $rootScope.$on('$stateChangeError', changeError);
+     var unregisterSuccess = $rootScope.$on('$stateChangeSuccess', changeSuccess);
++>>>>>>> 66af10e... Merge pull request #127 from ZitaNemeckova/about_modal
  
      function changeStart(event, toState, toParams, fromState, fromParams) {
        if (toState.data && !toState.data.requireUser) {
```

**Comment by chessbyte** [comment]
@ZitaNemeckova @chriskacerguis Please resolve git conflict locally, make PR against `euwe` branch and assign that PR to me.

**Comment by AllenBW** [comment]
Oh hai, i can get on this @ZitaNemeckova @chriskacerguis

**Comment by chessbyte** [comment]
Backported to Euwe via https://github.com/ManageIQ/manageiq-ui-self_service/pull/205
